pkg/provision/primitives/kubernetes.go

```diff
@@ -229,7 +229,11 @@ func (p *Provisioner) kubernetesInstall(ctx context.Context, name string, cpu ui
 		cmdline = fmt.Sprintf("%s k3os.server_url=https://%s:6443", cmdline, ipstring)
 	}
 	for _, key := range cfg.SSHKeys {
-		cmdline = fmt.Sprintf("%s ssh_authorized_keys=\"%s\"", cmdline, key)
+		trimmed := strings.TrimSpace(key)
+		if strings.ContainsAny(trimmed, "\"\n") {
```

Comment:
ssh keys are passed as a list so you can add multiple keys by providing them as items. Making them one item separated by new line will fail because of how it's parsed inside the vm. This is an [example](https://github.com/emmericp/initramfs-tools/blob/14a89aba6bd21f6e8af8177e2343a3b7a9bb804b/init#L75), so a new line would make the ssh key separated between two steps.